tests/general/include_config_source_test.go

```diff
@@ -0,0 +1,114 @@
+package tests
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/confmap"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+	"gopkg.in/yaml.v2"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+func TestCollectorProcessWithMultipleTemplateConfigs(t *testing.T) {
+	logCore, logs := observer.New(zap.DebugLevel)
+	logger := zap.New(logCore)
+	collector, err := testutils.NewCollectorProcess().
+		WithArgs("--config", path.Join(".", "testdata", "templated.yaml")).
+		WithLogger(logger).
+		Build()
+
+	require.NotNil(t, collector)
+	require.NoError(t, err)
+
+	err = collector.Start()
+	require.NoError(t, err)
+
+	_, ok := collector.(*testutils.CollectorProcess)
+	require.True(t, ok)
```

Comment:
```suggestion
```

I don't think these are necessary when not using the `CollectorProcess` directly.